test/dlc_tests/conftest.py

```diff
@@ -291,15 +312,207 @@ def ec2_instance_role_name(request):
 
 
 @pytest.fixture(scope="function")
-def ec2_instance_ami(request):
-    return request.param if hasattr(request, "param") else UBUNTU_18_BASE_DLAMI_US_WEST_2
+def ec2_instance_ami(request, region):
+    return (
+        request.param
+        if hasattr(request, "param")
+        else UBUNTU_18_BASE_DLAMI_US_EAST_1
+        if region == "us-east-1"
+        else UBUNTU_18_BASE_DLAMI_US_WEST_2
+    )
 
 
 @pytest.fixture(scope="function")
 def ei_accelerator_type(request):
     return request.param if hasattr(request, "param") else None
 
 
+@pytest.mark.timeout(300)
+@pytest.fixture(scope="function")
+def efa_ec2_instances(
```

Comment:
if there is same common code between this and `ec2_instances`, can we define common code and reuse it in both the methods?

Reply:
For now, there are some differences in the way the instance is launched for `ec2_instance` and `efa_ec2_instances`, such as a helper function to retry instance-launch, etc.

I'd like for us to have a separate PR to consolidate both functions, by using that retry+backoff functionality on instance launch for all tests, because this PR has already gotten complicated as-is.

In case the retry-backoff functionality is bad/flaky, the only tests affected right now will be the EFA tests, as opposed to all EC2 tests.

Reply:
ack